test/metric_value_benchmark/prometheus_test.go

```diff
@@ -26,27 +26,27 @@ var _ test_runner.ITestRunner = (*PrometheusTestRunner)(nil)
 //go:embed agent_configs/prometheus.yaml
 var prometheusConfig string
 
-const prometheusMetrics = `prometheus_test_untyped{include="yes"} 1
+const prometheusMetrics = `prometheus_test_untyped{include="yes",prom_type="untyped"} 1
 # TYPE prometheus_test_counter counter
-prometheus_test_counter{include="yes"} 1
+prometheus_test_counter{include="yes",prom_type="counter"} 1
 # TYPE prometheus_test_counter_exclude counter
-prometheus_test_counter_exclude{include="no"} 1
+prometheus_test_counter_exclude{include="no",prom_type="counter"} 1
 # TYPE prometheus_test_gauge gauge
-prometheus_test_gauge{include="yes"} 500
+prometheus_test_gauge{include="yes",prom_type="gauge"} 500
 # TYPE prometheus_test_summary summary
-prometheus_test_summary_sum{include="yes"} 200
-prometheus_test_summary_count{include="yes"} 50
-prometheus_test_summary{include="yes",quantile="0"} 0.1
-prometheus_test_summary{include="yes",quantile="0.5"} 0.25
-prometheus_test_summary{include="yes",quantile="1"} 5.5
+prometheus_test_summary_sum{include="yes",prom_type="summary"} 200
+prometheus_test_summary_count{include="yes",prom_type="summary"} 50
+prometheus_test_summary{include="yes",quantile="0",prom_type="summary"} 0.1
+prometheus_test_summary{include="yes",quantile="0.5",prom_type="summary"} 0.25
+prometheus_test_summary{include="yes",quantile="1",prom_type="summary"} 5.5
 # TYPE prometheus_test_histogram histogram
-prometheus_test_histogram_sum{include="yes"} 300
-prometheus_test_histogram_count{include="yes"} 75
-prometheus_test_histogram_bucket{include="yes",le="0"} 1
-prometheus_test_histogram_bucket{include="yes",le="0.5"} 2
-prometheus_test_histogram_bucket{include="yes",le="2.5"} 3
-prometheus_test_histogram_bucket{include="yes",le="5"} 4
-prometheus_test_histogram_bucket{include="yes",le="+Inf"} 5
+prometheus_test_histogram_sum{include="yes",prom_type="histogram"} 300
```

Comment:
Is the prom_type because of a bug in emf exporter? I was able to find prom_metric_type as a label in emf, and this worked for the prod agent.

Reply:
Yes. They add it as a label in the awsemfexporter but dont consider it when filtering for dimensions, hence it is a minor bug with the exporter.